### PR TITLE
Shell package

### PR DESCRIPTION
**Pull request by gimenete**
The shell opens a REPL with a `xata` global variable containing an initialized `XataClient` generated
for your current database and branch.

```shell
# The shell will use the env variables XATA_DATABASE_URL and XATA_API.
# You can define them in an `.env` file in the current directory.
npx @xata.io/shell
# Alternatively you can specify an env file
npx @xata.io/shell --env env.local
```

The shell uses the Node.js `repl` package with a couple of additions:

- Expressions that return a promise are resolved automatically.
- If you end an expression with `\t`, it will be printed with `console.table()`.

Example:

```js
// You can do
xata.db.teams.getMany() \t
// Which is equivalent to
console.table(await xata.db.teams.getMany())
```

TODO:
- use a stable version of codegen.
- In another PR I'll refactor the `scripts` we have to specify package versions to not duplicate them in each package.

### CODE REVIEW

File: client/src/schema/index.test.ts [hunk 1]
```diff
@@ -243,7 +246,7 @@ describe('request', () => {
       };
     });
 
-    await expect(users.getOne()).rejects.toThrowErrorMatchingInlineSnapshot(`"Not Found"`);
+    await expect(users.getOne()).rejects.toThrowErrorMatchingInlineSnapshot('"expected is not a function"');
```
Comment (fabien0102) [inline client/src/schema/index.test.ts:249]: This diff looks suspicious ^^

File: codegen/src/codegen.ts [hunk 2]
```diff
@@ -124,7 +123,7 @@ export async function generate({
       ${tables.map((table) => `"${table.name}": Repository<${getTypeName(table.name)}>;`).join('\n')}
     }> {
       constructor(options?: XataClientOptions) {
-        super({ databaseURL: "https://${config.workspaceID}.xata.sh/db/${config.dbName}", ...options}, links);
+        super({ databaseURL: "${databaseURL}", ...options}, links);
```
Comment (fabien0102) [inline codegen/src/codegen.ts:126]:
```suggestion
        super({ databaseURL, ...options}, links);
```

File: package.json [hunk 3]
```diff
@@ -4,8 +4,7 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "jest",
-    "test:watch": "jest --watch",
+    "test": "vitest",
```
Comment (fabien0102) [inline package.json:7]: 👌

File: codegen/src/codegen.ts (hunk 2 above)
Comment (SferaDev) [reply codegen/src/codegen.ts]: Actually no, this is the codegen, so we need to inline the ``databaseURL``

File: package.json (hunk 3 above)
Comment (SferaDev) [reply package.json]: The head branch was actually not correctly set, so the diff you were seeing was from https://github.com/xataio/client-ts/pull/173 😆

Comment (gimenete) [reply package.json]: Whops, my bad. Forgot to change the head branch when creating the PR.

**Comment by SferaDev** [comment]
image

**Comment by gimenete** [comment]
@SferaDev yeah, I'm aware. I'm trying to figure out why, because it is exported here https://github.com/xataio/client-ts/blob/main/codegen/src/index.ts#L2 and works locally, but not with npx 🤔

**Review by SferaDev — APPROVED** [review]

**Pull request merged**